addons/sale_management/models/sale_order.py

```diff
@@ -156,7 +160,7 @@ def action_confirm(self):
 
         for order in self:
             if order.sale_order_template_id and order.sale_order_template_id.mail_template_id:
-                self.sale_order_template_id.mail_template_id.send_mail(order.id)
+                order.sale_order_template_id.mail_template_id.with_user(SUPERUSER_ID).send_mail(order.id)
```

Comment:
You can only keep the change of order -> self
the Superuser_id part should already be fixed by the `if self.env.su:` part above (to verify)